modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```
@@ -743,71 +746,85 @@ private boolean isExpressionFunction(MetadataType metadataType)
+            }
+            else if (type.equals(Calendar.class))
+            {
+                Calendar calendar = Calendar.getInstance();
+                calendar.setTime(dateTime.toDate());
+                constructedValue = calendar;
+            }
+
+            if (constructedValue == null)
+            {
+                throw new IllegalArgumentException(format("Could not construct value of type '%s' from String '%s'", type.getName(), value));
+            }
+            else
+            {
+                value = constructedValue;
+            }
         }
-        else
+
+        if (value instanceof Date || value instanceof LocalDate || value instanceof LocalDateTime || value instanceof Calendar)
         {
-            Date date = doParseDate(value, CALENDAR_FORMAT, defaultValue);
-            Calendar calendar = Calendar.getInstance();
-            calendar.setTime(date);
-            return new StaticValueResolver(calendar);
+            return new StaticValueResolver(value);
         }
+
+        throw new IllegalArgumentException(format("Could not transform value of type '%s' to a valid date type", value != null ? value.getClass().getName() : "null"));
     }
 
-    private ValueResolver parseDate(Object value, DateType dateType, Object defaultValue)
+    private DateTime getParsedDateTime(String value)
     {
-        if (isExpression(value, parser))
+        try
         {
-            return new TypeSafeExpressionValueResolver((String) value, getType(dateType));
+            return ISODateTimeFormat.dateTimeParser().withOffsetParsed().parseDateTime(value);
         }
-        else
+        catch (DateTimeParseException e)
         {
-            return new StaticValueResolver(doParseDate(value, DATE_FORMAT, defaultValue));
+            throw new IllegalArgumentException(format("Could not parse value '%s' according to ISO 8601", value));
```

do not hide the original exception